openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -246,7 +246,7 @@ def _get_user_and_profile(username):
     """
     try:
         existing_user = User.objects.get(username=username)
-        existing_user_profile = UserProfile.objects.get(user=existing_user)
+        existing_user_profile, _ = UserProfile.objects.get_or_create(user=existing_user)
     except ObjectDoesNotExist:
```

Comment:
@ahsan-ul-haq I think @awaisdar001 has a good point. If we are using `UserProfile.objects.get_or_create()`, maybe we should move that statement (line 249) outside of the try and except clause. The except has nothing to do with getting profile anymore.